content/blog/tidymodels-2021-q2/index.Rmd

```diff
@@ -0,0 +1,232 @@
+
+```{r}
+butchered_plus <- function() {
+  some_junk_in_the_environment <- runif(1e6)
+  
+  churn_prep <- 
+    recipe(churn ~ voice_mail_plan + total_intl_minutes + 
+             total_day_minutes + total_eve_minutes + state, 
+           data = churn_train) %>%
+    step_dummy(all_nominal_predictors()) %>%
+    step_normalize(all_predictors()) %>%
+    prep()
+  
+  butcher(churn_prep)
+}
+```
+
+In the old version of butcher, we did not successfully remove all that extra stuff, and recipes were bigger than they needed to be:
+
+```{r, eval=FALSE}
+lobstr::obj_size(butcher(churn_prep))
+#> 1,835,512 B
+lobstr::obj_size(butchered_plus())
+#> 9,836,480 B
+```
+
+In the new version of butcher, we now successfully remove unneeded components from the recipe, so it is smaller:
+
+```{r, eval=FALSE}
+lobstr::obj_size(butcher(churn_prep))
+#> 1,695,352 B
+lobstr::obj_size(butchered_plus())
+#> 1,695,352 B
+```
+
+There are also `butcher()` methods for `workflow()` objects, so when you `butcher()` a modeling workflow, you remove everything not needed for prediction from both its estimated recipe *and* its trained model, making it as lightweight as possible for deployment.
+
+## SVMs and fast logistic regression with LiblineaR
+
+Unfortunately, the `"liquidSVM"` engine for support vector machine models that parsnip supported was deprecated in the latest release, because that package was removed from CRAN. We added a new engine in the same release that allows users to fit linear SVMs with the [parsnip model `svm_linear()`](https://parsnip.tidymodels.org/reference/svm_linear.html), as well as having another option for logistic regression. This new `"LiblineaR"` engine is based on the same C++ library that is shipped with scikit-learn. We'd like to thank the [maintainers of the LiblineaR R package](https://www.dnalytics.com/software/liblinear/) for all their help as we set up this integration.
+
+```{r}
+set.seed(234)
+churn_folds <- vfold_cv(churn_train, v = 5, strata = churn)
+
+liblinear_wf <-
+  logistic_reg(penalty = 0.2, mixture = 1) %>%
+  set_mode("classification") %>%
+  set_engine("LiblineaR")
+
+liblinear_wf <-
+  workflow() %>%
+  add_recipe(churn_rec) %>%
+  add_model(liblinear_wf)
+
+fit_resamples(liblinear_wf, resamples = churn_folds)
+```
+
+The `"LiblineaR"` engine for regularized logistic regression [can be very fast compared to the `"glmnet"` engine](https://twitter.com/juliasilge/status/1374029310764359681), even when we use a [sparse representation](https://www.tidyverse.org/blog/2020/11/tidymodels-sparse-support/). Check out [benchmarking code here](https://gist.github.com/juliasilge/26a43e5e68cf12842354e6652dfed688).
+
+## Post-processing your model predictions with probably and yardstick
```

Comment:
A musing: It's so weird to me that preprocessing has definitively reached closed form, but post-processing is still in compound.

Reply:
YES, so weird!